Source/Core/Common/VR/DolphinVR.cpp

```diff
@@ -0,0 +1,239 @@
+  SetControlState(0, GCPAD_B_BUTTON, rightController & xrButton_GripTrigger ? 1 : 0);
+  SetControlState(0, GCPAD_Z_BUTTON, rightController & xrButton_A ? 1 : 0);
+  SetControlState(0, GCPAD_DPAD_UP, rightController & xrButton_Up ? 1 : 0);
+  SetControlState(0, GCPAD_DPAD_DOWN, rightController & xrButton_Down ? 1 : 0);
+  SetControlState(0, GCPAD_DPAD_LEFT, rightController & xrButton_Left ? 1 : 0);
+  SetControlState(0, GCPAD_DPAD_RIGHT, rightController & xrButton_Right ? 1 : 0);
+  SetControlState(0, GCPAD_START_BUTTON, rightController & xrButton_RThumb ? 1 : 0);
+
+  // Left controller to WiiMote + Nunchuk
+  SetControlState(0, WIIMOTE_PLUS_BUTTON, leftController & xrButton_X ? 1 : 0);
+  SetControlState(0, WIIMOTE_MINUS_BUTTON, leftController & xrButton_Y ? 1 : 0);
+  SetControlState(0, WIIMOTE_HOME_BUTTON, leftController & xrButton_Back ? 1 : 0);
+  SetControlState(0, NUNCHUK_C_BUTTON, leftController & xrButton_Trigger ? 1 : 0);
+  SetControlState(0, NUNCHUK_Z_BUTTON, leftController & xrButton_GripTrigger ? 1 : 0);
+  SetControlState(0, NUNCHUK_STICK_X, IN_VRGetJoystickState(0).x);
+  SetControlState(0, NUNCHUK_STICK_Y, IN_VRGetJoystickState(0).y);
+
+  // Right controller to WiiMote
+  SetControlState(0, WIIMOTE_A_BUTTON, rightController & xrButton_Trigger ? 1 : 0);
+  SetControlState(0, WIIMOTE_B_BUTTON, rightController & xrButton_GripTrigger ? 1 : 0);
+  SetControlState(0, WIIMOTE_ONE_BUTTON, rightController & xrButton_A ? 1 : 0);
+  SetControlState(0, WIIMOTE_TWO_BUTTON, rightController & xrButton_B ? 1 : 0);
+  SetControlState(0, WIIMOTE_DPAD_UP, rightController & xrButton_Up ? 1 : 0);
+  SetControlState(0, WIIMOTE_DPAD_DOWN, rightController & xrButton_Down ? 1 : 0);
+  SetControlState(0, WIIMOTE_DPAD_LEFT, rightController & xrButton_Left ? 1 : 0);
+  SetControlState(0, WIIMOTE_DPAD_RIGHT, rightController & xrButton_Right ? 1 : 0);
+  SetControlState(0, WIIMOTE_IR_X, x);
+  SetControlState(0, WIIMOTE_IR_Y, y);
+}
+
+/*
+================================================================================
+
+VR app flow integration
+
+================================================================================
+*/
+
+bool IsVREnabled()
+{
+#ifdef OPENXR
+  return true;
+#else
+  return false;
+#endif
+}
+
+#ifdef ANDROID
+void InitVROnAndroid(void* vm, void* activity, const char* vendor, int version, const char* name)
+{
+  // Set platform flags
+  if (strcmp(vendor, "Pico") == 0)
+  {
+    VR_SetPlatformFLag(VR_PLATFORM_CONTROLLER_PICO, true);
+    VR_SetPlatformFLag(VR_PLATFORM_EXTENSION_INSTANCE, true);
+  }
+  else if ((strcmp(vendor, "Meta") == 0) || (strcmp(vendor, "Oculus") == 0))
+  {
+    VR_SetPlatformFLag(VR_PLATFORM_CONTROLLER_QUEST, true);
+    VR_SetPlatformFLag(VR_PLATFORM_EXTENSION_FOVEATION, false);
+    VR_SetPlatformFLag(VR_PLATFORM_EXTENSION_PERFORMANCE, true);
+  }
+
+  // Init VR
+  ovrJava java;
+  java.Vm = (JavaVM*)vm;
+  java.ActivityObject = (jobject)activity;
```

Comment:
Please use the correct types in the function declaration so you don't have to cast here.